great_expectations/rule_based_profiler/util.py

```diff
@@ -46,34 +39,72 @@ def get_validator(
             f"{expectation_suite_name}_{domain.id}_suite_{str(uuid.uuid4())[:8]}"
         )
 
-    return data_context.get_validator(
-        batch_request=batch_request,
-        create_expectation_suite_with_name=expectation_suite_name,
-    )
+    batch: Batch
+    if batch_list is None or all([batch is None for batch in batch_list]):
+        if batch_request is None:
+            return None
```

Comment:
Maybe consolidate on one line?

Reply:
@cdkini Cannot use the same line, because if the condition `if batch_request is None:` is False, we must not return, but continue on to the next statement (to obtain `batch_request` by building it differently).  Thanks.